### PR TITLE
update waitForSeconds argument type to meet other time accuracy requi…

### PR DESCRIPTION
**Pull request by zhuangshi23**

### CODE REVIEW

File: muduo/base/Condition.cc [hunk 1]
```diff
@@ -8,13 +8,22 @@
...
+  if (1e5 > nanoseconds){
+    nanoseconds = 1e5;
+  }
+
+  abstime.tv_sec += static_cast<time_t>(nanoseconds / kNanoSecondsPerSecond);
+  abstime.tv_nsec += static_cast<long>(nanoseconds % kNanoSecondsPerSecond);
+
   MutexLock::UnassignGuard ug(mutex_);
-  return ETIMEDOUT == pthread_cond_timedwait(&pcond_, mutex_.getPthreadMutex(), &abstime);
+  return ETIMEDOUT
```
Comment (chenshuo) [inline muduo/base/Condition.cc]: Please do not reformat code for no reason.

File: muduo/base/Condition.cc [hunk 2]
```diff
@@ -8,13 +8,22 @@
...
+bool muduo::Condition::waitForSeconds(double seconds)
 {
   struct timespec abstime;
   // FIXME: use CLOCK_MONOTONIC or CLOCK_MONOTONIC_RAW to prevent time rewind.
   clock_gettime(CLOCK_REALTIME, &abstime);
-  abstime.tv_sec += seconds;
+
+  const int64_t kNanoSecondsPerSecond = 1e9;
+  int64_t nanoseconds = static_cast<int64_t>(seconds * kNanoSecondsPerSecond);
+  if (1e5 > nanoseconds){
```
Comment (chenshuo) [inline muduo/base/Condition.cc]: if (nanoseconds < 1e5)
{

File: muduo/base/Condition.cc [hunk 3]
```diff
@@ -8,13 +8,22 @@
...
+    nanoseconds = 1e5;
+  }
+
+  abstime.tv_sec += static_cast<time_t>(nanoseconds / kNanoSecondsPerSecond);
+  abstime.tv_nsec += static_cast<long>(nanoseconds % kNanoSecondsPerSecond);
```
Comment (chenshuo) [inline muduo/base/Condition.cc]: This is wrong. tv_nsec could > 1e9.

File: muduo/base/Condition.cc (hunk 1 above)
Comment (zhuangshi23) [inline muduo/base/Condition.cc]: One line more than 80 characters? Seems like muduo NOT follow the Google programming specification strictly.

**Comment by zhuangshi23** [comment]
already update the PR, please review.

File: muduo/base/Condition.cc [hunk 4]
```diff
@@ -8,12 +8,22 @@
...
+
+  const int64_t kNanoSecondsPerSecond = 1e9;
+  int64_t nanoseconds = static_cast<int64_t>(seconds * kNanoSecondsPerSecond);
+  if (1e5 > nanoseconds)
+  {
+    nanoseconds = 1e5;
+  }
+
+  abstime.tv_sec += static_cast<time_t>(nanoseconds / kNanoSecondsPerSecond);
+  abstime.tv_nsec = (abstime.tv_nsec + nanoseconds) % kNanoSecondsPerSecond;
```
Comment (chenshuo) [inline muduo/base/Condition.cc]: This is wrong.
abstime = 1234.5
seconds = 0.5
your result: abstime = 1234.1
correct result: 1235.1

File: muduo/base/Condition.cc [hunk 5]
```diff
@@ -8,12 +8,22 @@
...
+bool muduo::Condition::waitForSeconds(double seconds)
 {
   struct timespec abstime;
   // FIXME: use CLOCK_MONOTONIC or CLOCK_MONOTONIC_RAW to prevent time rewind.
   clock_gettime(CLOCK_REALTIME, &abstime);
-  abstime.tv_sec += seconds;
+
+  const int64_t kNanoSecondsPerSecond = 1e9;
+  int64_t nanoseconds = static_cast<int64_t>(seconds * kNanoSecondsPerSecond);
+  if (1e5 > nanoseconds)
```
Comment (chenshuo) [inline muduo/base/Condition.cc]: if (nanoseconds < 1e5)

**Comment by zhuangshi23** [comment]
`bool muduo::Condition::waitForSeconds(double seconds)
{
  struct timespec abstime;
  // FIXME: use CLOCK_MONOTONIC or CLOCK_MONOTONIC_RAW to prevent time rewind.
  clock_gettime(CLOCK_REALTIME, &abstime);

  const int64_t kNanoSecondsPerSecond = 1e9;
  int64_t nanoseconds = static_cast<int64_t>(seconds * kNanoSecondsPerSecond);

  abstime.tv_sec += static_cast<time_t>((abstime.tv_nsec + nanoseconds) /
      kNanoSecondsPerSecond);
  abstime.tv_nsec = (abstime.tv_nsec + nanoseconds) % kNanoSecondsPerSecond;

  MutexLock::UnassignGuard ug(mutex_);
  return ETIMEDOUT == pthread_cond_timedwait(&pcond_, mutex_.getPthreadMutex(), &abstime);
}`

**Pull request closed without merging**